src/common/transformations/src/transformations/common_optimizations/group_normalization_fusion.cpp

```diff
@@ -0,0 +1,288 @@
+                return false;
+
+            // ensure that instance_norm_gamma will have shape compatible
+            // with group_norm parameters, i.e. 1D vector of shape (num_channels)
+            std::shared_ptr<ov::Node> instance_norm_gamma_1d_m = nullptr;
+            if (ov::shape_size(instance_norm_gamma.get_shape()) == 1) {
+                auto shape_1d_const_m = op::v0::Constant::create(element::i64, Shape{1}, {1});
+                instance_norm_gamma_1d_m =
+                    std::make_shared<ov::op::v1::Reshape>(instance_norm_gamma, shape_1d_const_m, true);
+            } else {
+                instance_norm_gamma_1d_m = std::make_shared<ov::op::v0::Squeeze>(instance_norm_gamma);
+            }
+            ov::OutputVector instance_norm_gamma_concat_inputs;
+            for (auto i = 0; i < channels_to_groups_ratio; i++)
+                instance_norm_gamma_concat_inputs.push_back(instance_norm_gamma_1d_m);
+            instance_norm_gamma_1d_m = std::make_shared<ov::op::v0::Concat>(instance_norm_gamma_concat_inputs, 0);
+            auto instance_norm_gamma_1d_out = instance_norm_gamma_1d_m->get_default_output();
+            auto instance_norm_gamma_1d_ps = instance_norm_gamma_1d_out.get_partial_shape();
+            if (instance_norm_gamma_1d_ps != expected_param_shape)
+                return false;
+
+            // group_norm_gamma /= instance_norm_gamma
+            group_norm_gamma_1d_m =
+                std::make_shared<ov::op::v1::Divide>(group_norm_gamma_1d_m, instance_norm_gamma_1d_m);
+
+            if (pattern_map.count(instance_norm_beta_m) > 0) {
+                // group_norm_beta -= group_norm_gamma * instance_norm_beta / instance_norm_gamma
+                auto group_norm_beta_corr_multiply_m =
+                    std::make_shared<ov::op::v1::Multiply>(group_norm_gamma_1d_m, instance_norm_beta_1d_m);
+                auto group_norm_beta_corr_divide_m =
+                    std::make_shared<ov::op::v1::Divide>(group_norm_beta_corr_multiply_m, instance_norm_gamma_1d_m);
+                group_norm_beta_1d_m =
+                    std::make_shared<ov::op::v1::Subtract>(group_norm_beta_1d_m, group_norm_beta_corr_divide_m);
+            }
+        } else {
+            if (pattern_map.count(instance_norm_beta_m) > 0) {
+                // group_norm_beta -= group_norm_gamma * instance_norm_beta
+                auto group_norm_beta_corr_multiply_m =
+                    std::make_shared<ov::op::v1::Multiply>(group_norm_gamma_1d_m, instance_norm_beta_1d_m);
+                group_norm_beta_1d_m =
+                    std::make_shared<ov::op::v1::Subtract>(group_norm_beta_1d_m, group_norm_beta_corr_multiply_m);
+            }
+        }
+
+        // we need to be able to cast mvn to MVN layer type
+        // in order to read actual epsilon value
+        auto mvn_out = pattern_map.at(mvn_m);
+        auto mvn = std::dynamic_pointer_cast<ov::op::v6::MVN>(mvn_out.get_node_shared_ptr());
```

Comment:
```suggestion
        auto mvn = ov::as_type_ptr<ov::op::v6::MVN>(mvn_out.get_node_shared_ptr());
```
Use OV cast, 
Or use `as_type<op::v6::MVN*>(mvn_out.get_node())` to not create shared pointers as it looks not required for mvn.,

the `mvn` can be null pointer or is validated before and here is not possible?

Reply:
Thanks, I've changed it to OV cast. My understanding is that there is no need for additional check, as it is set in the pattern as node of type "ov::op::v6::MVN" already.